src/_macosx.m

```diff
@@ -664,6 +659,7 @@ static CGFloat _get_device_scale(CGContextRef cr)
 FigureManager_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 {
     Window* window = [Window alloc];
+    ++FigureWindowCount;
```

Comment:
This belongs after the early-returns, or needs to be re-decremented before returning `NULL`. At the very least, this belongs after the check for invalid `window` on the next line.